aws/resource_aws_datasync_location_fsx_windows_file_system.go

```diff
@@ -0,0 +1,229 @@
+	return &schema.Resource{
+		Create: resourceAwsDataSyncLocationFsxWindowsFileSystemCreate,
+		Read:   resourceAwsDataSyncLocationFsxWindowsFileSystemRead,
+		Update: resourceAwsDataSyncLocationFsxWindowsFileSystemUpdate,
+		Delete: resourceAwsDataSyncLocationFsxWindowsFileSystemDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"fsx_filesystem_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"password": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringLenBetween(1, 104),
+			},
+			"user": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 104),
+			},
+			"domain": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 253),
+			},
+			"security_group_arns": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				MinItems: 1,
+				MaxItems: 5,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateArn,
+				},
+			},
+			"subdirectory": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 4096),
+			},
+			"tags": tagsSchema(),
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"creation_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDataSyncLocationFsxWindowsFileSystemCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).datasyncconn
+	fsxArn := d.Get("fsx_filesystem_arn").(string)
+
+	input := &datasync.CreateLocationFsxWindowsInput{
+		FsxFilesystemArn:  aws.String(fsxArn),
+		User:              aws.String(d.Get("user").(string)),
+		Password:          aws.String(d.Get("password").(string)),
+		SecurityGroupArns: expandStringSet(d.Get("security_group_arns").(*schema.Set)),
+		Tags:              keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().DatasyncTags(),
+	}
+
+	if v, ok := d.GetOk("subdirectory"); ok {
+		input.Subdirectory = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("domain"); ok {
+		input.Domain = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Creating DataSync Location Fsx Windows File System: %s", input)
+	output, err := conn.CreateLocationFsxWindows(input)
+	if err != nil {
+		return fmt.Errorf("error creating DataSync Location Fsx Windows File System: %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s#%s", aws.StringValue(output.LocationArn), fsxArn))
```

Comment:
Are we encoding the Fsx FS ARN into the resource ID here just so that we can get it in `resourceAwsDataSyncLocationFsxWindowsFileSystemRead()`, i.e. solely as use as the pass-through ID for resource import?
If so I would suggest moving the ID parsing code into a custom `Importer` (including `d.SetId(...)` and `d.Set("fsx_filesystem_arn", ...)`) and using the location ARN as the resource ID.

Reply:
done